dandiapi/api/views/asset.py

```diff
@@ -270,11 +271,12 @@ def paths(self, request, **kwargs):
         (to refer to the root folder) must be the empty string.
         """
         path_prefix: str = self.request.query_params.get('path_prefix') or ''
+        extended: bool = self.request.query_params.get('extended') or False
         # Enforce trailing slash
         if path_prefix and path_prefix[-1] != '/':
             path_prefix = f'{path_prefix}/'
         qs = self.get_queryset().filter(path__startswith=path_prefix).values()
 
-        return Response(Asset.get_path(path_prefix, qs))
+        return Response(Asset.get_path(path_prefix, qs, extended=extended))
```

Comment:
@dchiquito Does `Response` take advantage of a generator, in that e.g. would return first "page" whenever it is ready without waiting for all values to be generated?  then `get_path` could become a generator, yielding unsorted entries and it would be for UI to implement any sorting strategy it desires (folders first, size based, etc). This could then even reduce memory consumption by simply yielding individual assets results without memorizing them and only complimenting results at the end with folders' entries

Reply:
`Response` is unpaginated. You would need to explicitly create a new `DandiPagination`, use it (I forget the exact magic words), and return a `Response` with the results.

Reply:
forget that I have mentioned pagination since it seems it is some other layer somewhere since I see that `/paths` is supporting pagination according to interface in swagger.  The questions then are
- Can `Response` be provided a generator and not a list?
- Can we stop sorting here (at dandi-api level) since web-ui must be sorting anyways?

Reply:
the endpoint does not support pagination yet, the swagger interface is misleading at the moment.

```
curl -X GET "https://api.dandiarchive.org/api/dandisets/000026/versions/draft/assets/paths/?page_size=5&path_prefix=rawdata%2Fsub-I46%2Fses-SPIM%2F" -H  "accept: application/json"
```

Reply:
oh, filed https://github.com/dandi/dandi-api/issues/266 . Questions stand.